morl_baselines/common/pareto.py

```diff
@@ -66,8 +129,12 @@ def add(self, candidate, evaluation: np.ndarray):
         """
         self.evaluations.append(evaluation)
         self.individuals.append(deepcopy(candidate))
+
         # Non-dominated sorting
-        nd_candidates = get_non_dominated({tuple(e) for e in self.evaluations})
+        if self.convex_hull:
+            nd_candidates = filter_convex_dominated(self.evaluations)
+        else:
+            nd_candidates = filter_pareto_dominated(self.evaluations)
 
```

Comment:
I think the code here may be expecting a list, but filter is returning a np.ndarray?